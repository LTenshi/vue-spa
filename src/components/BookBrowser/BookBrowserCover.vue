<template>
  <tr>
    <th class="imageHeader">
        <img :src="bookCover" @click="openDetails" class="imageContainer">
    </th>
    <th class="bookName">
        <div>{{bookName}}</div>
        <div class="bookAuthor">{{bookAuthor}}</div>
    </th>
  </tr>
</template>

<script>
export default {
    props: {
        bookName: {
            type: String,
            required: true,
        },
        bookAuthor: {
            type: String,
            required: true,
        },
        bookCover: {
            type: String,
            required: true,
        },
    },
    methods: {
        openDetails() {
            this.$emit("openDetails");
        }
    }
}
</script>

<style>
.bookName, .bookAuthor {
    font-size: 13pt;
}

.bookName {
    vertical-align: top;
    padding: 20px;
    text-align: left;
}

.bookAuthor {
    font-weight: 400;
    color: rgb(175, 175, 175);
}

.imageContainer {
    max-width: 140px;
}
</style>