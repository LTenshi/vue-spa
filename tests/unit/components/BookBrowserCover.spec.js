import BookBrowserCover from "../../../src/components/BookBrowser/BookBrowserCover";
import { mount } from "@vue/test-utils";

describe("Book Browser Cover", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(BookBrowserCover, {
            props: {
               bookName: "Future",
               bookAuthor: "Dimitry Glukhovsky",
               bookCover: "imagineaurlhere"
            }
        });
    })

    it("Should match the snapshot", () => {
        expect(wrapper.element).toMatchSnapshot();
    })
    
    it('Should display the name', () => {
        expect(wrapper.html()).toContain("Future");
    });

    it('Should display the author', () => {
        expect(wrapper.html()).toContain("Dimitry Glukhovsky");
    });

    it('Should display the image', () => {
        expect(wrapper.html()).toContain("imagineaurlhere");
    });
    
});