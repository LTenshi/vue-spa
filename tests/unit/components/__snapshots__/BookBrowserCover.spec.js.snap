// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Browser Cover Should match the snapshot 1`] = `
<tr>
  <th
    class="imageHeader"
  >
    <img
      class="imageContainer"
      src="imagineaurlhere"
    />
  </th>
  <th
    class="bookName"
  >
    <div>
      Future
    </div>
    <div
      class="bookAuthor"
    >
      Dimitry Glukhovsky
    </div>
  </th>
</tr>
`;
